sdl_ios/SmartDeviceLink/SDLProtocol.m

```diff
@@ -237,22 +253,52 @@ - (void)processMessages {
 }
 
 - (void)sendHeartbeat {
-	SDLProtocolHeader *header = [SDLProtocolHeader headerForVersion:self.version];
+    SDLProtocolHeader *header = [SDLProtocolHeader headerForVersion:self.version];
     header.frameType = SDLFrameType_Control;
     header.serviceType = 0;
     header.frameData = SDLFrameData_Heartbeat;
-    header.sessionID = self.sessionID;
-
+    
     SDLProtocolMessage *message = [SDLProtocolMessage messageWithHeader:header andPayload:nil];
 
     [self sendDataToTransport:message.data withPriority:header.serviceType];
+    
+}
 
+- (void)sendRawDataStream:(NSInputStream *)inputStream withServiceType:(SDLServiceType)serviceType {
+    
+    SDLDataStreamHandlingDelegate *streamDelegate = [SDLDataStreamHandlingDelegate new];
+    streamDelegate.serviceType = serviceType;
+    streamDelegate.protocol = self;
+    objc_setAssociatedObject(inputStream, @"RetainedDelegate", streamDelegate, OBJC_ASSOCIATION_RETAIN);
```

Comment:
This whole idea of using associated objects to retain the stream delegate onto the input stream provided by developers is simply asking for bugs and heartache. If the developer doesn't realize they have to retain the input stream they pass in themselves (which they shouldn't have to), then this stream delegate will immediately be released and die.